### PR TITLE
Add secret-handshake

### PR DESCRIPTION
**Pull request by BNAndras**
#48in24 exercise

### CODE REVIEW

**Comment by BNAndras** [comment]
Not sure how to resolve the markdown lint.  It seems to not like 
```
~~~~exercism/note
If you aren't sure what binary is or how it works, check out [this binary tutorial][intro-to-binary].

[intro-to-binary]: https://medium.com/basecs/bits-bytes-building-with-binary-13cb4289aafa
~~~~
```

but that's an Exercism-specific piece of Markdown from upstream. Several other exercises have docs that use that markdown so this linter might be a problem.

**Comment by 0xNeshi** [comment]
Removed linting of autogenerated .md files, just merge the most recent changes

https://github.com/exercism/cairo/pull/144

File: exercises/practice/secret-handshake/.meta/example.cairo [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    for action in actions() {
+        if number.into() & (1 * pow(2, i)) != 0 {
+            results.append(action);
+        }
+        i += 1;
+    };
+
+    if number & 16 != 0 {
+        let mut reversed: Array<ByteArray> = array![];
+        let mut i = results.len();
```
Comment (0xNeshi) [inline exercises/practice/secret-handshake/.meta/example.cairo:14]: There's also a way to do this now using `while let` and popping from the back, removing the need for `i`.

```cairo
let mut span = results.span();
while let Option::Some(action) = span.pop_back() {
    reversed.append(action.clone());
}
```

You don't need to update this, just leaving it out there

File: config.json [hunk 2]
```diff
@@ -180,6 +180,14 @@
         "prerequisites": [],
         "difficulty": 2
       },
+      {
+        "slug": "secret-handshake",
+        "name": "Secret Handshake",
+        "uuid": "033c70bb-25cc-47ba-b4b7-92f0d012a5c8",
+        "practices": [],
+        "prerequisites": [],
+        "difficulty": 2
```
Comment (0xNeshi) [inline config.json:189]:
```suggestion
        "difficulty": 4
```
This doesn't seem like a "2" exercise, it's not as easy as e.g. `leap`. Will set it to 4

**Review by 0xNeshi — APPROVED** [review]
Good job!

P.S. I just increased the difficulty a bit, as "2" seemed too low

File: exercises/practice/secret-handshake/.meta/example.cairo (hunk 1 above)
Comment (BNAndras) [reply exercises/practice/secret-handshake/.meta/example.cairo:14]: Awesome.

**Pull request merged**